frontend/src/queries/settings.ts

```diff
@@ -1,24 +1,40 @@
+import type { Settings, SettingsKey } from "@fiberplane/fpx-types";
 import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { useMemo } from "react";
 
 const FPX_USER_SETTINGS_QUERY_KEY = "FPX_USER_SETTINGS";
 
 export function useFetchSettings() {
   return useQuery({
     queryKey: [FPX_USER_SETTINGS_QUERY_KEY],
-    queryFn: () => fetch("/v0/settings").then((r) => r.json()),
+    queryFn: async () => {
+      const response = await fetch("/v0/settings");
+      return (await response.json()) as Settings;
```

Comment:
nit: we could use the `SettingsSchema` here to validate the result as opposed to casting it, at the same time as we're controlling the API it's probably fine

Reply:
Good call! Updated 👍